### ISSUE TITLE
Missing argument for phpcr media admin

### ISSUE THREAD

**Issue by ElectricMaxxx** [sonata-project/SonataMediaBundle#1159]
#### Sonata packages

```bash
sonata-project/admin-bundle                3.10.3             The missing Symfony Admin Generator
sonata-project/block-bundle                3.2.0              Symfony SonataBlockBundle
sonata-project/cache                       1.0.7              Cache library
sonata-project/cache-bundle                2.3.1              This bundle provides caching services
sonata-project/classification-bundle       3.1.0              Symfony SonataClassificationBundle
sonata-project/core-bundle                 3.1.2              Symfony SonataCoreBundle
sonata-project/datagrid-bundle             2.2                Symfony SonataDatagridBundle
sonata-project/doctrine-extensions         1.0.2              Doctrine2 behavioral extensions
sonata-project/doctrine-orm-admin-bundle   3.1.2              Symfony Sonata / Integrate Doctrine ORM into the SonataAdminBundle
sonata-project/doctrine-phpcr-admin-bundle dev-master d6dc296 Symfony Sonata / Integrate Doctrine PHPCR into the SonataAdminBundle
sonata-project/easy-extends-bundle         2.1.10             Symfony SonataEasyExtendsBundle
sonata-project/exporter                    1.7.0              Lightweight Exporter library
sonata-project/intl-bundle                 2.2.4              Symfony SonataIntlBundle
sonata-project/media-bundle                3.2.0              Symfony SonataMediaBundle
sonata-project/notification-bundle         3.0.0              Symfony SonataNotificationBundle
sonata-project/seo-bundle                  2.0.2              Symfony SonataSeoBundle
sonata-project/translation-bundle          2.x-dev 500a6c4    SonataTranslationBundle

```

#### Symfony packages

```bash
symfony/assetic-bundle     v2.8.1 Integrates Assetic into Symfony2
symfony/monolog-bundle     2.12.0 Symfony MonologBundle
symfony/phpunit-bridge     v3.2.1 Symfony PHPUnit Bridge
symfony/polyfill-intl-icu  v1.3.0 Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-mbstring  v1.3.0 Symfony polyfill for the Mbstring extension
symfony/polyfill-php56     v1.3.0 Symfony polyfill backporting some PHP 5.6+ features to lower PHP versions
symfony/polyfill-php70     v1.3.0 Symfony polyfill backporting some PHP 7.0+ features to lower PHP versions
symfony/polyfill-util      v1.3.0 Symfony utilities for portability of PHP codes
symfony/security-acl       v3.0.0 Symfony Security Component - ACL (Access Control List)
symfony/swiftmailer-bundle v2.4.2 Symfony SwiftmailerBundle
symfony/symfony            v3.2.1 The Symfony PHP framework

```

#### PHP version

```bash
PHP 7.1.0-5+deb.sury.org~trusty+1 (cli) ( NTS )
Copyright (c) 1997-2016 The PHP Group
Zend Engine v3.1.0-dev, Copyright (c) 1998-2016 Zend Technologies
    with Zend OPcache v7.1.0-5+deb.sury.org~trusty+1, Copyright (c) 1999-2016, by Zend Technologies
    with Xdebug v2.5.0, Copyright (c) 2002-2016, by Derick Rethans
```

## Steps to reproduce

Install media-bundle, use PHPCR as db_driver

## Expected results

Admin is working fine.

## Actual results

```
Type error: Too few arguments to function Sonata\MediaBundle\Admin\BaseMediaAdmin::__construct(), 4 passed in /var/www/prberghoff/New/var/cache/dev/appDevDebugProjectContainer.php on line 9832 and exactly 5 expected
```
The service defintion of the phpcr media admin misses one argument, which the orm version has. Looking deeper shows, that the argument is a dependency into the classification bundle, where there is no phpcr (or other) db representation.

**Comment by greg0ire** [comment]
Can you provide a full stack trace?

> The service defintion of the phpcr media admin misses one argument

Shouldn't this be reported on the phpcr repo then?

**Comment by ElectricMaxxx** [comment]
The media admin has no own repo. All (ORM, Mongo and PHPCR) are defined here. Just have a look into the different admin service definitions, which goes into one base class constructor.

**Comment by greg0ire** [comment]
Oh indeed. Can you link to the relevant service definition?

**Comment by ElectricMaxxx** [comment]
The ORM admin definition:
https://github.com/sonata-project/SonataMediaBundle/blob/3.x/Resources/config/doctrine_orm_admin.xml#L24
The PHPC admin defintion:
https://github.com/sonata-project/SonataMediaBundle/blob/3.x/Resources/config/doctrine_phpcr_admin.xml#L24
the common constructor:
https://github.com/sonata-project/SonataMediaBundle/blob/3.x/Admin/BaseMediaAdmin.php#L41
(should look equal for mongo, right)

From my POV, we (@dbu or me from the CMF) have to give you some support on implementing an an phpcr category manager in the other bundle and use it here, right?

**Comment by greg0ire** [comment]
The third argument was introduced in 0dbc19f09fff787f1846494b67feec88b7ce88ad , more than 2 years ago (!)

**Comment by dbu** [comment]
seems we are lacking tests. and the cmf-sandbox was locked on older versions of sonata that did not have the break. if it can be made optional that seems to be the best solution to me.

**Comment by greg0ire** [comment]
> if it can be made optional that seems to be the best solution to me

@ElectricMaxxx could you try working on that? I had a quick glance at the base admin and it seems doable.

**Comment by greg0ire** [comment]
> seems we are lacking tests

The coverage is ~ 35% indeed :sweat:

**Comment by ElectricMaxxx** [comment]
Yea. I will have a look into it tonight. But I think it is more just fixing the admin. A short fix would be to make the complete category manager optional, which should have been the first solution when implementing it for ORM. A long and better solution would be to implement it for PHPCR on classification bundle. I would prefer last solution.  

Another thing that is shown by that issue: user acceptance of phpcr media with newer versions. I would like to push both (sonata and PHPCR) a little bit more into public. The tasks that are solved here are awesome and almost every file related symfony app would need that. And from my CMF way of thinking those information should be persisted in a tree. Unfortunately our media-bundle is currently unmaintained and has from my POV some overlap with our resource component (Puli). But let's have a talk on that after our 2.0 maybe with somebody of sonata too. I will try to get our current media-bundle working again (I do need it for a personal project).

**Comment by greg0ire** [comment]
> I would prefer last solution.

Ideally both should be done IMO, if the category can be considered optional. That would make implementing a new persistence backend easier. Plus it looks like [the mongoDB admin would benefit from such a fix too](https://github.com/sonata-project/SonataMediaBundle/blob/3.x/Resources/config/doctrine_mongodb_admin.xml#L23)

**Comment by ElectricMaxxx** [comment]
Just realized the the argument is  fixed on master, so can you please tag a minor release for that?
For the change of classification-bundle i opened: https://github.com/sonata-project/SonataClassificationBundle/issues/262

**Comment by greg0ire** [comment]
Master cannot receive any release, it is for the next major. If you want that change soon, you need to backport it on the stable branch, in a BC manner.

**Comment by ElectricMaxxx** [comment]
i think https://github.com/sonata-project/SonataMediaBundle/commit/dbbf8499f1f5769db13b20404288318a6700461f#diff-745f47255ede2dd7e1b9234f82409eb3 is a bug-fix on a feature implemented before, so it should be released as a patch.

**Comment by greg0ire** [comment]
If this diff is BC, then it should have been on the stable branch not on master. We have never released anything from master, and if when we do, the tag will be 4.0.0 . You can cherry-pick this commit on 3.x an create a PR from it if you think it is BC.

**Comment by ElectricMaxxx** [comment]
argh .. to many conflicts in unknown code ... I will have to use an other solution then, thanks.

**Comment by greg0ire** [comment]
You can try porting it manually I guess, without cherry-pick

**Comment by jordisala1991** [comment]
Can you check on the latest SonataMediaBundle is this is still an issue? @ElectricMaxxx

**Comment by jordisala1991** [comment]
This is fixed since #1201
